pkg/kubectl/cmd/util/printing.go

```diff
@@ -31,7 +31,11 @@ func AddPrinterFlags(cmd *cobra.Command) {
 	cmd.Flags().StringP("output", "o", "", "Output format. One of: json|yaml|template|templatefile|wide.")
 	cmd.Flags().String("output-version", "", "Output the formatted object with the given version (default api-version).")
 	cmd.Flags().Bool("no-headers", false, "When using the default output, don't print headers.")
-	cmd.Flags().StringP("template", "t", "", "Template string or path to template file to use when -o=template or -o=templatefile.  The template format is golang templates [http://golang.org/pkg/text/template/#pkg-overview]")
+	// template shorthand -t is deprecated to support -t for --tty
+	// TODO: remove template flag shorthand -t
+	cmd.Flags().StringP("template", "t", "", "Template string or path to template file to use when -o=template or -o=templatefile.  The template format is golang templates [http://golang.org/pkg/text/template/#pkg-overview].")
+	// TODO: uncomment this line after MarkShorthandDeprecated() is ready to be used
+	// cmd.Flags().MarkShorthandDeprecated("template", "please use --template")
```

Comment:
@deads2k By setting usage message "`please use --template`", users will see "`Flag shorthand -t has been deprecated, please use --template`" when they use `-t`. Is it good enough?